apps/aecore/lib/aecore/structures/signed_tx.ex

```diff
@@ -24,13 +25,33 @@ defmodule Aecore.Structures.SignedTx do
   use ExConstructor
 
   @spec is_coinbase?(SignedTx.t()) :: boolean()
-  def is_coinbase?(tx) do
-    tx.data.from_acc == nil && tx.signature == nil
+  def is_coinbase?(%{data: %{from_acc: key}, signature: signature}) do
+    key == nil && signature == nil
   end
 
   @spec is_valid?(SignedTx.t()) :: boolean()
-  def is_valid?(tx) do
-    tx.data.value >= 0 && tx.data.fee >= 0 && Keys.verify_tx(tx)
+  def is_valid?(%{data: data, signature: signature}) do
+    data.value >= 0 && data.fee >= 0 &&
+      Signing.verify(Serialization.pack_binary(data), signature, data.from_acc)
```

Comment:
Maybe move pack_binary to Signing.verify, it would be easier to change it and it won't have to be written every time something is signed

Reply:
This function is implemented in the `elxir-wallet` repository so I don't think it's a good idea to change it's base functionality for this. It takes binary as the first argument and I'd like to keep it this way. This way the function is more generic and not bouned to some specific implementation.

Reply:
ok, lets keep it like this